src/Providers/Stripe/Card/Service.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace PodPoint\Payments\Providers\Stripe\Card;
+
+use PodPoint\Payments\Card;
+use PodPoint\Payments\Card\Service as ServiceInterface;
+use PodPoint\Payments\Exception;
+use PodPoint\Payments\Providers\Stripe\Card\Exception as CardException;
+use PodPoint\Payments\Providers\Stripe\Token as StripeToken;
+use PodPoint\Payments\Token;
+use Stripe\SetupIntent;
+use Stripe\PaymentMethod;
+use Stripe\Customer as StripeCustomer;
+
+class Service implements ServiceInterface
+{
+    /**
+     * Tries attach a card to a customer using the Stripe SDK.
+     *
+     * @param Token $cardToken
+     * @param Token $customerToken
+     *
+     * @return Card
+     */
+    public function attach(Token $cardToken, Token $customerToken): Card
+    {
+        if ($cardToken->type !== StripeToken::PAYMENT_METHOD && $customerToken->type !== StripeToken::CUSTOMER) {
+            throw new \Exception('You need to provide a valid card Token and customer Token');
+        }
```

Comment:
I don't think we should worry about this as we are invoking this method we should just ensure we are invoking it right